src/proposals/dto/update-proposal.dto.ts

```diff
@@ -1,10 +1,119 @@
-import { PartialType } from "@nestjs/swagger";
-import { Exclude } from "class-transformer";
-import { IsOptional } from "class-validator";
-import { CreateProposalDto } from "./create-proposal.dto";
+import { ApiProperty, ApiTags, PartialType } from "@nestjs/swagger";
+import { Type } from "class-transformer";
+import {
+  IsArray,
+  IsDateString,
+  IsEmail,
+  IsOptional,
+  IsString,
+  ValidateNested,
+} from "class-validator";
+import { OwnableDto } from "../../common/dto/ownable.dto";
+import { MeasurementPeriodClass } from "../schemas/measurement-period.schema";
+import { CreateMeasurementPeriodDto } from "./create-measurement-period.dto";
 
-export class UpdateProposalDto extends PartialType(CreateProposalDto) {
+@ApiTags("proposals")
+export class UpdateProposalDto extends OwnableDto {
+  @ApiProperty({
+    type: String,
+    required: false,
+    description: "Email of principal investigator.",
+  })
   @IsOptional()
-  @Exclude()
-  readonly proposalId: string;
+  @IsEmail()
+  readonly pi_email?: string;
+
+  @ApiProperty({
+    type: String,
+    required: false,
+    description: "First name of principal investigator.",
+  })
+  @IsOptional()
+  @IsString()
+  readonly pi_firstname?: string;
+
+  @ApiProperty({
+    type: String,
+    required: false,
+    description: "Last name of principal investigator.",
+  })
+  @IsOptional()
+  @IsString()
+  readonly pi_lastname?: string;
+
+  @ApiProperty({
+    type: String,
+    required: true,
+    description: "Email of main proposer.",
+  })
+  @IsEmail()
+  readonly email: string;
+
+  @ApiProperty({
+    type: String,
+    required: false,
+    description: "First name of main proposer.",
+  })
+  @IsOptional()
+  @IsString()
+  readonly firstname?: string;
+
+  @ApiProperty({
+    type: String,
+    required: false,
+    description: "Last name of main proposer.",
+  })
+  @IsOptional()
+  @IsString()
+  readonly lastname?: string;
+
+  @ApiProperty({
+    type: String,
+    required: true,
+    description: "The title of the proposal.",
+  })
+  @IsString()
+  readonly title: string;
+
+  @ApiProperty({
+    type: String,
+    required: false,
+    description: "The proposal abstract.",
+  })
+  @IsOptional()
+  @IsString()
+  readonly abstract?: string;
+
+  @ApiProperty({
+    type: Date,
+    required: false,
+    description: "The date when the data collection starts.",
+  })
+  @IsOptional()
+  @IsDateString()
+  readonly startTime?: Date;
+
+  @ApiProperty({
+    type: Date,
+    required: false,
+    description: "The date when data collection finishes.",
+  })
+  @IsOptional()
+  @IsDateString()
+  readonly endTime?: Date;
+
+  @ApiProperty({
+    type: MeasurementPeriodClass,
```

Comment:
I think this should be a `CreateMeasurementPeriodDto` rather than a `MeasurementPeriodClass`

Reply:
You are correct, thank you very much!